test/SparqlAntlrParserTest.cpp

```diff
@@ -1279,3 +1280,32 @@ TEST(SparqlParser, GroupGraphPattern) {
   expectGroupGraphPatternFails("{ SERVICE <foo> { } }");
   expectGroupGraphPatternFails("{ SERVICE SILENT ?bar { } }");
 }
+
+namespace {
+template <typename Exception = ParseException>
+void expectRDFLiteralFails(const string& input) {
+  EXPECT_THROW(parseRdfLiteral(input), Exception) << input;
+}
+}  // namespace
```

Comment:
I think we have this "...fails" functions now pretty often. And in most cases we always use the default argument `ParseException`. So this can become something like

``` template<auto F, typename Exception = ParseException>
auto expectParsingFails = [](const string& input) {
  EXPECT_THROW(F(input), Exception) << input;
};
```

When you use it always with the default argument, you can then simply write
`auto expectRDFLiteralFails = expectParsingFails<parseRdfLiteral>`, which can also be directly inside the parsing functions. If you sometimes need to specify the type of the exceptions, then you need to write

```
template <typename Exception = ParseException>
auto expectBlaFails = expectParsingFails<parseBla, Exception>
```

(unfortunately this has to be outside of a function, but maybe you define all of these together in one anonymous namespace, right next to the `parseBla` macros at the beginning of the file. ) Then we have cleaned up the "actual" tests and the testHelpers.

Reply:
`makeExpectParsingFails` probably has to return a lambda so having template parameters in the returned lambda is not easily possible. But because the parser now only throws `ParseException` this shouldn't be a problem.

Reply:
I don't understand what you mean by `makeExpectParsingFails`. I think my above approach should work and cover all cases. If you always throw and assert for ParseExceptions, then the simple, local case always works.